### PR TITLE
Eager mode examples and README

### PR DESCRIPTION
**Pull request by makslevental**
Add a simple and ResNet18 example for Eager mode. Also add a bit of description in the README

### CODE REVIEW

File: examples/eager_mode.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+from torch_mlir.eager_mode.torch_mlir_tensor import TorchMLIRTensor
+
+a = TorchMLIRTensor(torch.randn(5, requires_grad=True))
+b = TorchMLIRTensor(torch.randn(5, requires_grad=True))
+
+c = a + b
+d = a * b
+e = c / d
+
+e.sum().backward()
```
Comment (silvasean) [inline examples/eager_mode.py:17]: print the sum.
Also print out some gradients after the backward call too to show off how seamless this all is :)

(also, random question: does TorchMLIRTensor also cause the backward graph to execute through Torch-MLIR?)

**Review by silvasean — CHANGES_REQUESTED** [review]

File: examples/eager_mode.py (hunk 1 above)
Comment (makslevental) [reply examples/eager_mode.py:17]: >also, random question: does TorchMLIRTensor also cause the backward graph to execute through Torch-MLIR

yup - it's like magic (the `grad_fn` gets attached to the `TorchMLIRTensor` since it's a subclass of `torch.Tensor` and then during backwards `__torch_dispatch__` fires just like during the forward pass).

File: examples/eager_mode.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+print("PyTorch grad b: ", torch_b.grad)
+
+a = TorchMLIRTensor(torch_a)
+b = TorchMLIRTensor(torch_b)
+
+c = a + b
+d = a * b
+e = c / d
+loss = e.sum()
+print("TorchMLIR loss: ", loss)
```
Comment (silvasean) [inline examples/eager_mode.py:30]: nit: Torch-MLIR with a `-`.

**Review by silvasean — APPROVED** [review]
